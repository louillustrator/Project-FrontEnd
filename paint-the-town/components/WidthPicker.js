import React, { Component } from 'react';
import { StyleSheet, View, Text } from "react-native";
import Slider from 'react-native-slider';

class WidthPicker extends Component {
    state={
        width: 0
    }
    render() {
        return (
          <View style={styles.container}>
          
          <Slider 
          style={styles.slider}
          value={this.state.width}
          step={1}
          minimumValue={0}
          maximumValue={10}
          onValueChange={width => this.setState({width})}
          onSlidingComplete={this.slidingComplete}
          />
          <Text style={styles.text}>{this.state.width}</Text>
          </View>
        );
    }

    componentDidMount() {
        this.setState({
            width: this.props.currentWidth
        })
    }

    slidingComplete = () => {
        this.props.setShowSlider(false);
        this.props.setWidth(this.state.width)
    }
}

const styles = StyleSheet.create({
    container: {
        height: 150,
        justifyContent: 'center',    
        marginLeft: 30,
        marginRight: 30,
        zIndex: 10,
        backgroundColor: 'rgba(52, 52, 52, 0.0)'
    },
    text: {
      fontSize: 30,
      textAlign: 'center',
    },
    slider: {
        width: 200
      },
  });

export default WidthPicker;