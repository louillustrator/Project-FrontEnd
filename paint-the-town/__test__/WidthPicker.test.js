import React from "react";
import { View } from "react-native";
import { mount } from 'enzyme';
import renderer from "react-test-renderer";
import WidthPicker from "../components/WidthPicker";


describe('Polylines', () => {
    let props;
    let state;
    beforeEach(() => {
        props = {
            currentWidth: 1,
            setShowSlider: jest.fn(),
            setWidth: jest.fn()
        };
        state = {
            width: 0
        };
    });
    it('renders without crashing', () => {
        mount(<WidthPicker {...props} />);
    });
    it("renders correctly", () => {
        const tree = renderer.create(<WidthPicker {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
