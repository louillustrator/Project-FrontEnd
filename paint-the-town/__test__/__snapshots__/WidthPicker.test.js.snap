// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Polylines renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(52, 52, 52, 0.0)",
      "height": 150,
      "justifyContent": "center",
      "marginLeft": 30,
      "marginRight": 30,
      "zIndex": 10,
    }
  }
>
  <View
    animationType="timing"
    onLayout={[Function]}
    onSlidingComplete={[Function]}
    onValueChange={[Function]}
    step={1}
    style={
      Array [
        Object {
          "height": 40,
          "justifyContent": "center",
        },
        Object {
          "width": 200,
        },
      ]
    }
    thumbTouchSize={
      Object {
        "height": 40,
        "width": 40,
      }
    }
    value={1}
  >
    <View
      onLayout={[Function]}
      renderToHardwareTextureAndroid={true}
      style={
        Array [
          Object {
            "backgroundColor": "#b3b3b3",
          },
          Object {
            "borderRadius": 2,
            "height": 4,
          },
          undefined,
        ]
      }
    />
    <View
      renderToHardwareTextureAndroid={true}
      style={
        Object {
          "backgroundColor": "#3f3f3f",
          "borderRadius": 2,
          "height": 4,
          "opacity": 0,
          "position": "absolute",
          "width": 0,
        }
      }
    />
    <View
      onLayout={[Function]}
      renderToHardwareTextureAndroid={true}
      style={
        Object {
          "backgroundColor": "#343434",
          "borderRadius": 10,
          "height": 20,
          "opacity": 0,
          "position": "absolute",
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
          "width": 20,
        }
      }
    />
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      renderToHardwareTextureAndroid={true}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {},
        ]
      }
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 30,
        "textAlign": "center",
      }
    }
  >
    1
  </Text>
</View>
`;
